spec/unit/resolver_spec.rb

```diff
@@ -396,15 +396,25 @@ module Pod
         e.message.should.match(/`AFNetworking \(> 2\)` required by `Podfile`/)
       end
 
-      it 'raises if no such version of a dependency exists' do
+      # it 'raises if no such version of a dependency exists' do
+      #   podfile = Podfile.new do
+      #     platform :ios
+      #     pod 'AFNetworking', '3.0.1'
+      #   end
+      #   resolver = Resolver.new(config.sandbox, podfile, empty_graph, SourcesManager.all)
+      #   e = lambda { resolver.resolve }.should.raise Informative
+      #   e.message.should.match(/Unable to find a specification for the following requirement/)
+      #   e.message.should.match(/AFNetworking.*= 3\.0\.1/)
+      # end
+
+      it 'test which should raise with no such dependency error' do
         podfile = Podfile.new do
           platform :ios
           pod 'AFNetworking', '3.0.1'
         end
         resolver = Resolver.new(config.sandbox, podfile, empty_graph, SourcesManager.all)
         e = lambda { resolver.resolve }.should.raise Informative
-        e.message.should.match(/Unable to satisfy the following requirements/)
-        e.message.should.match(/`AFNetworking \(= 3.0.1\)` required by `Podfile`/)
```

Comment:
Current implementation actually satisfies the above two lines. What it should do is to satisfy the line below. I didn't add a check about the failing pod because this spec was to show the erroneous case. Correct spec will be the commented out lines.